### PR TITLE
feat: Support 1Password Connect as a backend

### PR DESCRIPTION
**Pull request by lucasmarshall**
### Description
Add support for 1Password Connect servers as a backend

**Fixes:** #235

### Checklist
Please make sure that your PR fulfills the following requirements:
- [X] Reviewed the guidelines for contributing to this repository
- [X] The commit message follows the [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [X] Tests for the changes have been updated
- [X] Docs have been added / updated
- [ ] Optional. My organization is added to USERS.md.

### Type of Change

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [X] New tests
- [ ] Build/CI related changes
- [X] Documentation content changes
- [ ] Other (please describe)

### CODE REVIEW

File: docs/backends.md
Comment (werne2j) [inline docs/backends.md:525]: If `OP_CONNECT_TOKEN` and `OP_CONNECT_HOST` are env variables that 1password connect will read by default, https://github.com/1Password/connect-sdk-go#environment-variables, i would rather we strip the `AVP_` and just support the native env vars like we do with Vault and others.

This would also require adding `op_connect` to this list: https://github.com/argoproj-labs/argocd-vault-plugin/blob/main/pkg/config/config.go#L44

Comment (lucasmarshall) [reply docs/backends.md:525]: Sure, that's reasonable. Will push a new commit shortly

File: pkg/config/config_test.go
Comment (werne2j) [inline pkg/config/config_test.go:160]: Can you also add a case for a using the one password backend with a missing variable in the TestNewConfigMissingParameter test

Comment (lucasmarshall) [reply pkg/config/config_test.go:160]: Sure

Comment (lucasmarshall) [reply pkg/config/config_test.go:160]: Just realized I neglected to push the additional test I wrote to the correct remote. Fixed now.

**Review by werne2j — APPROVED** [review]
Looks good, will let @jkayani take a look before merging

File: go.mod
Comment (jkayani) [inline go.mod:58]: This is a direct dependency isn't it? Wondering why it's marked as `// indirect` then

File: pkg/config/config.go
Comment (jkayani) [inline pkg/config/config.go:219]: Wondering if this is necessary, assuming that the `OP_CONNECT` vars are the ones expected by the SDK; omitting this check and deferring to the SDK means we can support any auth methods this backend supports going forward with no code changes, as long as we set all `OP_CONNECT` prefixed config keys in the environment (which your change to add `op_connect` to the backend prefixes slice does). Not a big deal, just a thought

File: docs/config.md
Comment (jkayani) [inline docs/config.md:76]: If `OP_CONNECT` prefixed env vars are what the SDK expects I don't think we need to explicitly call them out here, the note above this table should point users to the right place:

> We support all the backend specific environment variables each backend's SDK will accept (e.g, `VAULT_NAMESPACE`, `AWS_REGION`, etc). Refer to the [specific backend's documentation](./backends) for details.

Otherwise, we should keep them here

File: docs/backends.md
Comment (jkayani) [inline docs/backends.md:516]: Does this backend support versioning? I don't see anything to suggest it does in the code, in which case we should add an explicit note here that specified versions will be ignored

**Review by jkayani — COMMENTED** [review]
Code, tests and docs look great! Thanks for the work @lucasmarshall. I'd like to resolve at least the dependency question since the OP SDK should be marked as a direct dependency (thinking about build issues if dependencies aren't categorized correctly). The other things aren't big deals but I leave it to you to make any changes if you see fit.

File: docs/backends.md
Comment (lucasmarshall) [reply docs/backends.md:516]: It does not. I will add a note

File: go.mod
Comment (werne2j) [inline go.mod:55]: Sorry, one small thing. I would like this to be up in the above require block. Seems a little odd to have another require block with just one package.

**Review by werne2j — CHANGES_REQUESTED** [review]

File: go.mod
Comment (lucasmarshall) [reply go.mod:55]: Sure. `go mod tidy` put this here. Not sure why.

**Pull request merged**
